enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,272 @@
+---
+title: advertise-version-details
+authors:
+  - "@dhellmann"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@spadgett, console team"
+  - "@wking, upgrades team"
+  - "@sdodson, upgrades team"
+  - "@LalatenduMohanty, upgrades team"
+  - "@soltysh, workloads team"
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2022-07-29
+last-updated: 2022-07-29
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/USHIFT-233
+  - https://bugzilla.redhat.com/show_bug.cgi?id=1850656
+---
+
+# Advertise Version Details
+
+## Summary
+
+We have several use cases for allowing authenticated users to discover
+whether they are communicating with OCP or MicroShift, including the
+version of the product. We already have the ClusterVersion resource
+for OCP, but that requires special privileges to read and is not
+present in a MicroShift deployment. This enhancement describes a
+standard that both platforms can use to share platform and version
+information with API callers.
+
+## Motivation
+
+Users configuring applications to deploy on different versions of
+OpenShift may need to adjust their configuration to work differently
+on OCP and MicroShift. Ideally this is not needed, but in practice it
+is useful because of different APIs being available and different
+constraints being in place. It is possible for a cluster admin to read
```

Comment:
Having constraints tied to a product/version lookup table seems brittle.  And it feels like it will overlap with [the Infrastructure topology properties][1].  But it may be a reasonable stopgap while folks work on a more structured declaration system, and it's possible that we never have enough maintenance burden to push us into developing structured declarations for these particular constraints :shrug:

[1]: https://github.com/openshift/api/blob/9368b92de0aba00c7220273fd51875034d7ded7e/config/v1/types_infrastructure.go#L84-L103

Reply:
I added a more explicit use case about this below.